packages/typespec-client-generator-core/src/http.ts

```diff
@@ -292,75 +297,90 @@ function addContentTypeInfoToBodyParam(
   return diagnostics.diagnostics;
 }
 
+/**
+ * Generate TCGC Http parameter type, `httpParam` or `location` should be provided at least one
+ * @param context
+ * @param param TypeSpec param for the http parameter
+ * @param operation
+ * @param httpParam TypeSpec Http parameter type
+ * @param location Location of the http parameter
+ * @returns
+ */
 export function getSdkHttpParameter(
   context: TCGCContext,
-  type: ModelProperty,
+  param: ModelProperty,
   operation?: Operation,
+  httpParam?: HttpOperationParameter,
   location?: "path" | "query" | "header" | "body"
-): [SdkHttpParameter, readonly Diagnostic[]] {
+): [SdkHttpParameter | void, readonly Diagnostic[]] {
```

Comment:
On which case would `void` be returned? What does it mean for a parameter to be void?

What do you expect emitter do on this `void`?